dhis-2/dhis-support/dhis-support-external/src/main/java/org/hisp/dhis/external/conf/ConfigurationKey.java

```diff
@@ -149,7 +149,16 @@
     AUDIT_ENABLED( "system.audit.enabled", Constants.TRUE, false ),
     TRACKER_IMPORT_PREHEAT_CACHE_ENABLED( "tracker.import.preheat.cache.enabled", Constants.ON, false ),
     ENABLE_OAUTH2_AUTHORIZATION_SERVER( "oauth2.authorization.server.enabled", Constants.ON, false ),
-    ENABLE_JWT_OIDC_TOKEN_AUTHENTICATION( "oidc.jwt.token.authentication.enabled", Constants.OFF, false );
+    ENABLE_JWT_OIDC_TOKEN_AUTHENTICATION( "oidc.jwt.token.authentication.enabled", Constants.OFF, false ),
+    CSP_NONCE_ENABLED( "csp.nonce.enabled", Constants.ON, true ),
+    CSP_UPGRADE_INSECURE_ENABLED( "csp.upgrade.insecure.enabled", Constants.OFF, true ),
+    CSP_HEADER_VALUE( "csp.header.value", "prefetch-src 'self'; "
```

Comment:
Is this value getting slightly long for dhis.conf property? Could we perhaps instead "hard-code" the default values/domains somewhere else, and then have this property for extra domains only which are then appended to the default? My point is that is implementations want to add extra domains they won't have to include all this text.

Reply:
Yes, I agree this is getting too big. I can move this into a separate class.